src/db.rs

```diff
@@ -288,11 +296,12 @@ impl<'a> DBRawIterator<'a> {
     fn new_cf(
         db: &DB,
         cf_handle: &ColumnFamily,
-        readopts: &ReadOptions,
+        readopts: ReadOptions,
     ) -> Result<DBRawIterator<'a>, Error> {
         unsafe {
             Ok(DBRawIterator {
                 inner: ffi::rocksdb_create_iterator_cf(db.inner, readopts.inner, cf_handle.inner),
+                _readopts: readopts,
```

Comment:
Could we clone `readopts` here in favour to leave public API as is ?

Reply:
I can clone it if we want to avoid API changes. I'm just curious what the plan is for the public API changes? It seems that you just merged https://github.com/rust-rocksdb/rust-rocksdb/pull/383 which already changed some of these APIs. Maybe we could make some public API changes in the next release anyway?

Reply:
@wqfish the problem with current API is that readopts are not clonable, so you're forced to construct them every time you want to use them for a new iterator.
We should try to avoid unnecessary API changes whenever possible, because each API change is an additional work for our users.

Reply:
Now that I see `ReadOptions` doesn't implement `Clone`, I don't feel this can be done in a backward compatible way. If we still let `readopts` own the buffer, we either need to take ownership of `readopts` like the current PR, or add some lifetime restriction to ensure that `readopts` outlives the iterator. Both would break some existing code and it seems that the changes caused by taking ownership of `readopts` might actually be minimal compared to other options.

Also note that the APIs for `Snapshot` currently take `readopts` by value, so this could be a bit consistency improvements as well.

Thoughts?

Reply:
What happened with this discussion?
At least making `ReadOptions` clonnable would be very desirable - right now this API got broken if one just has `ReadOptions` passed by reference / stored once in the wrapper struct.